inline-c/src/Language/C/Types/Parse.hs

```diff
@@ -293,11 +298,14 @@ data TypeSpecifier
   | Struct CIdentifier
   | Enum CIdentifier
   | TypeName CIdentifier
+  | Template CIdentifier [TypeSpecifier]
+  | TemplateConst String
```

Comment:
Right now this seems to only parse integral const expressions. That's OK, but can you document this, linking to some documentation on what is the full range of const template argument that you can have?